### PR TITLE
Do not load root `babel.config.js` in tests

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | 
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

While working on the native ESM Babel build (wip implementation: https://github.com/nicolo-ribaudo/babel/tree/native-esm), I had to add some `configFile: false` to our tests (you can see them for `babel-core` at https://github.com/nicolo-ribaudo/babel/commit/5d15dc703d3d741375a7a8a25a23d32cfdf52adf) because they were accidentally loading the top-level `babel.config.js` file.

This didn't cause problems so far because our `babel.config.js` has an `ignore` entry for `packages/*/test/fixtures`, but it caused problems for the ESM tests because even if a file is ignored all the plugins/presets are still resolved and `require`'d.

Also, tests should be isolated so it's better to make sure that they don't accidentally load an unrelated config anyway.

cc @jridgewell If I recall correctly, AMP uses our test runner plugins. I would consider this PR a bugfix, but would it cause problems to AMP?

### CODE REVIEW

**Review by hzoo — APPROVED** [review]
yeah I'd think we'd only want to apply the config/options from options.json? Do we need to include babelrc: false as well?

File: packages/babel-helper-transform-fixture-test-runner/src/index.ts [hunk 1]
```diff
@@ -27,11 +27,22 @@ const cachedScripts = new QuickLRU({ maxSize: 10 });
 const contextModuleCache = new WeakMap();
 const sharedTestContext = createContext();
 
+// We never want our tests to accidentally load the root
+// babel.config.js file, so we disable config loading by
+// default. Tests can still set `configFile: true | string`
+// to re-enable config loading.
+function transformWithoutConfigFile(code, opts) {
+  return babel.transform(code, {
+    configFile: false,
```
Comment (nicolo-ribaudo) [inline packages/babel-helper-transform-fixture-test-runner/src/index.ts:36]:
```suggestion
    configFile: false,
    babelrc: false,
```

**Review by jridgewell — APPROVED** [review]

**Review by jridgewell — APPROVED** [review]
We actually handle this by loading an empty config:

1. https://github.com/ampproject/amphtml/blob/d8e414a727aede5093c6e60100dc30e7c7c488f8/babel.config.js#L36
2. https://github.com/ampproject/amphtml/blob/d8e414a727aede5093c6e60100dc30e7c7c488f8/build-system/babel-config/empty-config.js

**Comment by nicolo-ribaudo** [comment]
Oh well, good to know that this solve a real problem!

**Pull request merged**